tensorflow/python/keras/utils/data_utils.py

```diff
@@ -293,7 +293,7 @@ def _hash_file(fpath, algorithm='sha256', chunk_size=65535):
   if (algorithm == 'sha256') or (algorithm == 'auto' and len(hash) == 64):
     hasher = hashlib.sha256()
   else:
-    hasher = hashlib.md5()
+    hasher = hashlib.sha1()
```

Comment:
From @fchollet 

> The behavior of `_hash_file` needs to stay backwards compatible since the output is used to match cached files that are already written (i.e. it interacts with the real world, it is not an internal implementation detail). Unless there are critical concerns, we should revert this line.